### ISSUE TITLE
How to set up a pure JS project

### ISSUE THREAD

**Issue by ckuchimanchi** [sourcegraph/javascript-typescript-langserver#533]
Hi,

The LSP works fine on a single JS file -- it identifies "go to definition" etc. 

But what if I have a bunch of js files in a folder and want to work on a type that's defined in another file. Go to definition doesn't work there. I think for it to happen, LSP should support a "project definition" file like tsconfig.json (for typescript) in javascript projects as well -- 

A quick search reveals, visual studio code does support such a file - jsconfig.json

Not sure if such a thing is supported by LSP? If so, does this implementation adhere to it? I tried creating that file but looks like it's not working. 

Can you please advise?

Thanks!

**Comment by felixfbecker** [comment]
The language server already reads jsconfig.json

**Comment by ckuchimanchi** [comment]
Thanks for the quick response @felixfbecker. Can you please tell me if the below jsconfig.json is appropriate?

`{    
    "include": [
        "snc/**/*.js"
    ]
}
`
I'm trying to include all .js files under 'snc' folder in all subdirectories.
 
It doesn't seem to be working.  

Go to definition isn't working f the class/type is present in another file. **However**, If I open the file manually and then go back and do go to definition, it works!

Any help would be appreciated. Thank you.

**Comment by ckuchimanchi** [comment]
Hi - Can anyone please help me with this?

**Comment by felixfbecker** [comment]
I don't know what could be the cause here

**Comment by jmartin84** [comment]
I've been seeing a similar issue.

**Comment by orzechowskid** [comment]
yep, I'm seeing this with emacs as well (specifically, eglot-mode plus flymake-mode).  Setting my tsconfig.json to

```JSON
{
    "compilerOptions": {
        "allowJs": true
    },
    "exclude": [
        "node_modules"
    ],
    "include": [
        "src/**/*"
    ]
}
```

did not help; flymake still displays TypeScript error messages.

**Comment by cjk** [comment]
@orzechowskid What kind of error-messages are you seeing?

The following `jsconfig.json` file is working for me (my js-files reside under `./src`):

``` json
{
  "compilerOptions": {
    "target": "es2017",
    "module": "es2015",
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": true,
    "noEmit": true,
    "checkJs": true,
    "allowJs": true,
    "lib": ["es2017"],
    "baseUrl": "./src"
  },
  "exclude": ["node_modules"],
  "include": ["src/**/*"]
}
```

**Comment by karolyi** [comment]
I have a JS project developed for the frontend part of my application with webpack, and this project doesn't give any code completions to it. I'm investigating the issue for more than an hour now, it's getting frustrating.

jsconfig.json doesn't seem to help.
